File: fanny_string.py
# Fanny is given a string along with the string which contains 
# single character x. She has to remove the character x from the
#  given string. Help her write a function to remove all occurrences of x 
# character from the given string.

# Input Specification:

# input1: input string s 

# input2: String containing any character x

# Output Specification:

# String without the occurrence of character x

# Example 1:

# input1: welcome to mettl 

# input2: l



# Output: wecome to mett
def removechar(s,c) :
    counts = s.count(c)
    s=list(s)
    while counts:
        s.remove(c)
        counts -= 1
    s = ''.join(s)
    print(s)

# drivers code
if __name__ == '__main__':
    s="welcome to mettle"
    removechar(s,'l')

#time complexity O(n) where n is length of string
# space complexity O(1)

File: fanny_string.cpp
#include<iostream>
using namespace std;
string removech(string str,char ch){
    int len = str.length();
    string s = "";
    for(int i=0;i<len;i++){
        char c = str[i];
        if(c == ch)
            continue;
        else
            s=s+c;
    }

    return s;
}
int main(){
    string str;
    char ch;
    getline (cin,str);
    cin >> ch;
    cout <<removech(str,ch);
    return 0;
}

//#time complexity O(n) where n is length of string
// space complexity O(1)